.github/workflows/build.yml

```diff
@@ -94,31 +89,8 @@ jobs:
         if: matrix.php >= 8.0
         run: ./bin/behat -fprogress --strict --tags=@php8
 
-  static-analysis:
-    name: Static analysis
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: Setup PHP
-        uses: shivammathur/setup-php@v2
-        with:
-          php-version: 8.1
-          ini-values: "zend.exception_ignore_args=Off"
-          coverage: none
-
-      # until psalm is updated to v5 which is compatible with Symfony 7
-      - name: Force symfony version for psalm compatibility
-        run: |
-          composer config --global --no-plugins allow-plugins.symfony/flex true &&
-          composer global require symfony/flex
-
-      - name: Install dependencies
-        env:
-          SYMFONY_REQUIRE: 5.4.*
-        run: composer update
-
       - name: Run Psalm
+        if: matrix.php != 8.4 && matrix.composer-mode != 'lowest'
```

Comment:
We can't run Psalm on PHP 8.4 because it is not yet compatible with this PHP version. Also, we don't want to run it with the lower version of dependencies because it shows some errors which have been fixed in later versions